source/hdev/Camera/Basic/CaptureHDRLoop.hdev

```diff
@@ -85,43 +87,20 @@
 <l>    get_dict_tuple (json_dict, 'Settings', settings)</l>
 <l>    get_dict_tuple (settings, 'Acquisitions', acquisitions)</l>
 <c>    </c>
-<c>    </c>
-<c>    * Getting settings for first acqusition</c>
-<l>    get_dict_tuple (acquisitions, 0, firstAcquisition)</l>
-<l>    get_dict_tuple (firstAcquisition, 'Acquisition', acquisition)</l>
-<l>    get_dict_tuple (acquisition, 'ExposureTime', exposure_time)</l>
-<l>    set_framegrabber_param (AcqHandle, 'ExposureTime', exposure_time)</l>
-<l>    get_dict_tuple (acquisition, 'Aperture', aperture)</l>
-<l>    set_framegrabber_param (AcqHandle, 'Aperture', aperture)</l>
-<l>    get_dict_tuple (acquisition, 'Brightness', brightness)</l>
-<l>    set_framegrabber_param (AcqHandle, 'Brightness', brightness)</l>
-<l>    get_dict_tuple (acquisition, 'Gain', gain)</l>
-<l>    set_framegrabber_param (AcqHandle, 'Gain', gain)</l>
-<c>    </c>
-<c>    * Getting settigns for second acqusition</c>
-<l>    get_dict_tuple (acquisitions, 1, secondAcquisition)</l>
-<l>    get_dict_tuple (secondAcquisition, 'Acquisition', acquisition)</l>
-<l>    get_dict_tuple (acquisition, 'ExposureTime', exposure_time)</l>
-<l>    set_framegrabber_param (AcqHandle, 'ExposureTime', exposure_time)</l>
-<l>    get_dict_tuple (acquisition, 'Aperture', aperture)</l>
-<l>    set_framegrabber_param (AcqHandle, 'Aperture', aperture)</l>
-<l>    get_dict_tuple (acquisition, 'Brightness', brightness)</l>
-<l>    set_framegrabber_param (AcqHandle, 'Brightness', brightness)</l>
-<l>    get_dict_tuple (acquisition, 'Gain', gain)</l>
-<l>    set_framegrabber_param (AcqHandle, 'Gain', gain)</l>
-<c>    </c>
-<c>    * Getting settings for third acqusition</c>
-<l>    get_dict_tuple (acquisitions, 2, thirdAcquisition)</l>
-<l>    get_dict_tuple (thirdAcquisition, 'Acquisition', acquisition)</l>
-<l>    get_dict_tuple (acquisition, 'ExposureTime', exposure_time)</l>
-<l>    set_framegrabber_param (AcqHandle, 'ExposureTime', exposure_time)</l>
-<l>    get_dict_tuple (acquisition, 'Aperture', aperture)</l>
-<l>    set_framegrabber_param (AcqHandle, 'Aperture', aperture)</l>
-<l>    get_dict_tuple (acquisition, 'Brightness', brightness)</l>
-<l>    set_framegrabber_param (AcqHandle, 'Brightness', brightness)</l>
-<l>    get_dict_tuple (acquisition, 'Gain', gain)</l>
-<l>    set_framegrabber_param (AcqHandle, 'Gain', gain)</l>
-<c>    </c>
+<c>    * Getting acquisition settings</c>
+<l>    for Index := 0 to |AllKeys| - 1 by 1</l>
+<l>        get_dict_tuple (acquisitions, Index, tempAcquisition)</l>
+<l>        get_dict_tuple (tempAcquisition, 'Acquisition', acquisition)</l>
+<l>        get_dict_tuple (acquisition, 'ExposureTime', exposure_time)</l>
```

Comment:
Especially this I wonder about. 
The tuple that gets created from the json file is:
Acquisitions:
- 0:
    * Acquisition:
        - ExposureTime........
- 1:
    * Acquisition:
        - ExposureTime......
        
This makes the demand for creating a middle_tuple to extract the "0" and "1"-value as a tempVariable. Do you guys have any other solution for this?

Reply:
You should be able to call one function inside the other using slightly different syntax. Check this: https://github.com/zivid/zivid-halcon-samples/blob/196c6b4b79f44bd1dc00601a813eadaaeacf6f5d/source/hdev/Procedures/get_first_available_zivid_device.hdvp#L18-L24 - maybe you get a hint how to try that.

Reply:
From https://www.mvtec.com/doc/halcon/13/en/tuple_split.html see:
For `tuple_split( : : String, Separator : Substrings)`
HDevelop In-line Operation
HDevelop provides an in-line operation for tuple_split, which can be used in an expression in the following syntax:
`Substrings := split(String, Separator)`

Reply:
Would this work?
`acquisition := get_dict_tuple('Acquisition', get_dict_tuple (acquisitions, Index) )`

Reply:
Maybe I misunderstand, but I didn't think functions in Halcon returns a value. They are usually assigned inside the functions and the function itself returns success or failure? https://www.mvtec.com/doc/halcon/1811/en/get_dict_tuple.html

If that is the case then this will not work?

Reply:
My bad on the initial comment. We are creating a dict, not a tuple